cli/sync_service/object.go

```diff
@@ -5,104 +5,170 @@ import (
 	"fmt"
 	"github.com/open-horizon/anax/cli/cliutils"
 	"github.com/open-horizon/anax/config"
-	"github.com/open-horizon/anax/exchange"
 	"github.com/open-horizon/anax/i18n"
 	"github.com/open-horizon/edge-sync-service/common"
 	"net/http"
 	"os"
 	"path"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
 )
 
 type MMSObjectInfo struct {
-	Definition   common.MetaData             `json:"definition"`
+	ObjectID     string                      `json:"objectID,omitempty"`
+	ObjectType   string                      `json:"objectType,omitempty"`
+	Definition   *common.MetaData            `json:"definition,omitempty"`
 	Destinations []common.DestinationsStatus `json:"destinations,omitempty"`
+	ObjectStatus string                      `json:"objectStatus,omitempty"`
 }
 
-// Display the object metadata for a given object in the MMS.
-func ObjectList(org string, userPw string, objType string, objId string, details bool) {
+// Display the object metadata for given flags in the MMS.
+func ObjectList(org string, userPw string, objType string, objId string, destPolicy string, dpService string, dpPropertyName string, dpUpdateTimeSince string, destType string, destId string, withData string, expirationTimeBefore string, long bool, details bool) {
 	// get message printer
 	msgPrinter := i18n.GetMessagePrinter()
 
 	if userPw == "" {
 		cliutils.Fatal(cliutils.CLI_INPUT_ERROR, msgPrinter.Sprintf("must specify exchange credentials to access the model management service."))
 	}
 
-	if details && objId == "" {
-		cliutils.Fatal(cliutils.CLI_INPUT_ERROR, msgPrinter.Sprintf("must specify object ID when requesting object details."))
-	}
-
 	// Set the API key env var if that's what we're using.
 	cliutils.SetWhetherUsingApiKey(userPw)
 
-	// If object ID is omitted, query all objects of the given type.
-	if !details {
-		objectList := new(exchange.ObjectDestinationPolicies)
-		urlPath := path.Join("api/v1/objects/", org, objType, "?all_objects=true")
+	var objectsMeta []common.MetaData
+
```

Comment:
Please write a comment describing the rules of which parameters are valid with other parameters, and also describing what parameter validation is being done.